src/utils/Firebase/firebase.js

```diff
@@ -36,23 +31,153 @@ class Firebase {
 
   doSignOut = () => this.auth.signOut();
 
-  // EXAMPLE ENDPOINT
-  getLesson = () => this.db.collection("custom_lesson");
-
-  //Gets last mastered lesson of a given student.
-  getLastMastered = deploymentAccountId =>
+  //Gets the most recent lesson that a student has completed
+  getLastMasteredLesson = deploymentAccountId =>
     this.db
       .doc(`deployment_account/${deploymentAccountId}/`)
       .get()
-      .then(documentSnapshot => {
-        documentSnapshot.get("profile.glenLearn.lastMasteredLesson");
-      });
+      .then(deploymentDoc => {
+        const lastMasteredLesson = deploymentDoc.get(
+          "profile.glenLearn.lastMasteredLesson"
+        );
+        return new Promise((resolve, reject) => {
+          if (deploymentDoc) {
+            resolve(lastMasteredLesson);
+          } else {
+            reject(Error("Deployment doc does not exist"));
+          }
+        });
+      })
+      .catch(error => console.log("Error getting student account: ", error));
 
-  //Data access APIs
-  customLessons = adminAccountId =>
+  getTeacherCustomLessons = adminAccountId =>
     this.db
       .collection(`custom_lesson`)
-      .where("adminAccountId", "==", adminAccountId);
+      .where("adminAccountId", "==", adminAccountId)
+      .get()
+      .then(querySnapshot => {
+        const customLessons = querySnapshot.docs.map(doc => doc.data());
+        return new Promise((resolve, reject) => {
+          if (querySnapshot) {
+            resolve(customLessons);
+          } else {
+            reject(Error("No custom lessons found."));
+          }
+        });
+      })
+      .catch(error => console.log("Error getting custom lessons: ", error));
+
+  getStudentCustomLessons = deploymentAccountId =>
```

Comment:
also instead of student use deploymentaccount

Reply:
do you mean getDeploymentAccountCustomLessons instead of StudentCustomLessons?